### PR TITLE
Refactor open date range to use std::optional

### PR DESCRIPTION
**Pull request by promag**
Use `std::nullopt` for open date range instead of `TransactionFilterProxy::MIN_DATE` and `TransactionFilterProxy::MAX_DATE`.

### CODE REVIEW

**Review by monstro805 — APPROVED** [review]

**Review by monstro805 — APPROVED** [review]

**Comment by hebasto** [comment]
Concept ACK

**Review by hebasto — APPROVED** [review]
ACK a7837ba08bc10c9aaca6f28d9c2be9b8430323f5, I have reviewed the code and it looks OK, I agree it can be merged.

**Comment by hebasto** [comment]
nit: include `<optional>` to the `src/qt/transactionview.cpp` as `std::nullopt` are use there.

**Review by hebasto — APPROVED** [review]
re-ACK 4830f4912a538600e9e9133442e9f8d929006630, only missed header included since my [previous](https://github.com/bitcoin-core/gui/pull/354#pullrequestreview-682108182) review.

File: src/qt/transactionfilterproxy.cpp [hunk 1]
```diff
@@ -46,8 +39,8 @@ bool TransactionFilterProxy::filterAcceptsRow(int sourceRow, const QModelIndex &
         return false;
 
     QDateTime datetime = index.data(TransactionTableModel::DateRole).toDateTime();
-    if (datetime < dateFrom || datetime > dateTo)
-        return false;
+    if (dateFrom && datetime < *dateFrom) return false;
```
Comment (Talkless) [inline src/qt/transactionfilterproxy.cpp:42]: `std::optional` does support comparison without dereferencing (see (26) in https://en.cppreference.com/w/cpp/utility/optional/operator_cmp) but your variant is better as `operator<` would repeat `if (opt)` check you already (rightly) did.

**Review by Talkless — APPROVED** [review]
tACK 4830f4912a538600e9e9133442e9f8d929006630, tested on Debian Sid, filtering seems to work as expected.

**Pull request merged**